pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/MultipleTreesBuilder.java

```diff
@@ -115,13 +126,38 @@ public MultipleTreesBuilder(@Nullable List<StarTreeIndexConfig> indexConfigs, bo
     }
   }
 
+  private StarTreeIndexSeparator getSeparator()
+      throws Exception {
+    if (!_metadataProperties.containsKey(MetadataKey.STAR_TREE_COUNT)) {
+      return null;
+    }
+    try {
+      _separatorTempDir = new File(_segmentDirectory, StarTreeV2Constants.EXISTING_STAR_TREE_TEMP_DIR);
+      FileUtils.forceMkdir(_separatorTempDir);
+      FileUtils.copyFileToDirectory(
+          new File(_segmentDirectory, StarTreeV2Constants.INDEX_FILE_NAME), _separatorTempDir, false);
+      FileUtils.copyFileToDirectory(
+          new File(_segmentDirectory, StarTreeV2Constants.INDEX_MAP_FILE_NAME), _separatorTempDir, false);
```

Comment:
Can we move the file instead of copying? We can move the index files and only remove the metadata fields

Reply:
That's a valid point, but I didn't move the files as I wanted to use `StarTreeBuilderUtils.removeStarTrees` (to avoid duplicating the startree index cleanup logic), and that was throwing exception upon missing startree index files. Let me know your thoughts on this.

Reply:
Moving file is much faster than copying file in most FS. Let's use move and directly clean up the fields in `_metadataProperties` by doing `_metadataProperties.subset(StarTreeV2Constants.MetadataKey.STAR_TREE_SUBSET).clear();`. This way we can also avoid reloading the `_metadataProperties`